testing/scenario_app/ios/Scenarios/ScenariosUITests/StatusBarTest.m

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "StatusBarTest.h"
+
+@implementation StatusBarTest
+
+- (void)setUp {
+  [super setUp];
+  self.continueAfterFailure = NO;
+
+  self.application = [[XCUIApplication alloc] init];
+  self.application.launchArguments = @[ @"--tap-status-bar" ];
+  [self.application launch];
+}
+
+- (void)testTapStatusBar {
+  if (@available(iOS 13, *)) {
+    XCUIApplication* systemApp =
+        [[XCUIApplication alloc] initWithBundleIdentifier:@"com.apple.springboard"];
+    XCUIElement* statusBar = [systemApp.statusBars firstMatch];
+    if (statusBar.isHittable) {
+      [statusBar tap];
+    } else {
+      XCUICoordinate* coordinates = [statusBar coordinateWithNormalizedOffset:CGVectorMake(0, 0)];
+      [coordinates tap];
+    }
+  } else {
+    [[self.application.statusBars firstMatch] tap];
+  }
+
+  XCUIElement* addTextField = self.application.textFields[@"PointerChange.add"];
```

Comment:
oh ha. Though can't you just put the channel handler you had in the AppDelegate in this test itself? Then you can just [self waitForExpectations] against its own listener like AppLifecycleTests?

Reply:
You can't do that.  AppLifecycleTests are unit tests and run in the same process as the app.  UI tests run as a separate process, I'd have to do IPC to communicate between the processes.  There is a builtin mechanism to do IPC for UI elements though so that's what I'm using here.

Reply:
Doh, you're right, this is a XCUITest